Modules/Office365DSC/DSCResources/MSFT_EXOAntiPhishRule/MSFT_EXOAntiPhishRule.psm1

```diff
@@ -0,0 +1,487 @@
+    Write-Verbose "$( Get-PSSession -ErrorAction SilentlyContinue | Where-Object Name -eq 'ExchangeOnline' | Out-String)"
+    try
+    {
+        $AntiPhishRules = Get-AntiPhishRule
+    }
+    catch
+    {
+        Close-SessionsAndReturnError -ExceptionMessage $_.Exception
+    }
+
+    $AntiPhishRule = $AntiPhishRules | Where-Object Identity -eq $Identity
+    if (-NOT $AntiPhishRule)
+    {
+        Write-Verbose "AntiPhishRule $($Identity) does not exist."
+        $result = @{
+            Ensure             = 'Absent'
+            GlobalAdminAccount = $GlobalAdminAccount
+            Identity           = $Identity
+        }
+        return $result
+    }
+    else
+    {
+        $result = @{
+            Ensure = 'Present'
+        }
+        foreach ($KeyName in ($PSBoundParameters.Keys | Where-Object {$_ -ne 'Ensure'}) )
+        {
+            if ($null -ne $AntiPhishRule.$KeyName)
+            {
+                $result += @{
+                    $KeyName = $AntiPhishRule.$KeyName
+                }
+            }
+            else
+            {
+                $result += @{
+                    $KeyName = $PSBoundParameters[$KeyName]
+                }
+            }
+
+        }
+        if ('Enabled' -eq $AntiPhishRule.State)
+        {
+            # Accounts for Get-AntiPhishRule returning 'State' instead of 'Enabled' used by New/Set
+            $result.Enabled = $true
+        }
+        else
+        {
+            $result.Enabled = $false
+        }
+
+        Write-Verbose "Found AntiPhishRule $($Identity)"
+        Write-Verbose "Get-TargetResource Result: `n $($result | Out-String)"
+        return $result
+    }
+
+}
+
+
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $AntiPhishPolicy,
+
+        [Parameter()]
+        [System.String]
+        $Comments,
+
+        [Parameter()]
+        [System.Boolean]
+        $Enabled = $true,
+
+        [Parameter()]
+        [ValidateSet('Present', 'Absent')]
+        [System.String]
+        $Ensure = 'Present',
+
+        [Parameter()]
+        [System.String[]]
+        $ExceptIfRecipientDomainIs = @(),
+
+        [Parameter()]
+        [System.String[]]
+        $ExceptIfSentTo = @(),
+
+        [Parameter()]
+        [System.String[]]
+        $ExceptIfSentToMemberOf = @(),
+
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $GlobalAdminAccount,
+
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $Identity,
+
+        [Parameter()]
+        [uint32]
+        $Priority,
+
+        [Parameter()]
+        [System.String[]]
+        $RecipientDomainIs = @(),
+
+        [Parameter()]
+        [System.String[]]
+        $SentTo = @(),
+
+        [Parameter()]
+        [System.String[]]
+        $SentToMemberOf = @()
+
+    )
+    function BuildAntiPhishParams
```

Comment:
Again, do not embed functions

Reply:
Functions moved to Office365DSCUtil module